Detectors/CPV/calib/include/CPVCalibration/NoiseCalibrator.h

```diff
@@ -51,9 +51,12 @@ class NoiseCalibrator final : public o2::calibration::TimeSlotCalibration<o2::cp
   void setPedEfficiencies(std::vector<float>* pedEffs) { mPedEfficiencies.reset(pedEffs); }
   void setDeadChannels(std::vector<int>* deadChs) { mDeadChannels.reset(deadChs); }
   void setHighPedChannels(std::vector<int>* highPeds) { mHighPedChannels.reset(highPeds); }
+  void setPersistentBadChannels(std::vector<int>* persBadChs) { mPersistentBadChannels.reset(persBadChs); }
```

Comment:
hi @shahor02 
yes, I also was thinking about this issue. However it worked fine. Thanks for pointing this out.

Reply:
It worked only because not all DPL tasks are calling destructor, also, it should fail in case of updating the object by more recent one...

Reply:
I see. Now changed to simple pointers. But there is one thing that I do not understand a bit:
when new object is fetched and old is deleted then I need somehow propagate new one into noise calibrator. But currently I access ccdb object only once at the beginning:
```
    static bool isConfigFetched = false;
    if (!isConfigFetched) {
      pc.inputs().get<std::vector<float>*>("pedeffs");
      pc.inputs().get<std::vector<int>*>("persbadchs");
      pc.inputs().get<std::vector<int>*>("deadchs");
      pc.inputs().get<std::vector<int>*>("highpeds");
      ...
      isConfigFetched = true;
    }
```
What to do in this case? Maybe use shared_ptr in order to be sure that old object is not deleted?
Actually this objects are not expected to change during noise and physics runs. However things must be done correctly in order to avoid unexpected problems in future.

Reply:
no, shared_ptr is also deleted once only 1 instance remains. Currently the objects in the `if (!isConfigFetched) {` are not supposed to change during the run lifetime. If they do, then you should remove this if and in e.g. finaliseCCDB pass them to the calibrator.